### PR TITLE
Update all deps including graphql-code-generator

### PR DESCRIPTION
**Pull request by cdupuis**
This updates graphql-code-generator to latest version which needed a couple of changes:

* now using the api to generate because otherwise we would need to have `codegen.yml` file in each repo 👎 
* I had to patch the web packed handlebar template to get the old bahavior back that generates optional properties on the generated types. Without that change, we have many compile errors in sdm and sdm-core then further.

Should we move forward with this for now? The motivation for the upgrade was a security vulnerability
on one of the old transitive dependencies. WDYT?

### CODE REVIEW

File: bin/gql-gen.ts [hunk 1]
```diff
@@ -40,53 +40,54 @@ async function main(): Promise<void> {
         const cwd = process.cwd();
         const lib = libDir(cwd);
 
+        // patch up codegen template to create optional properties
+        const codegenTemplate = path.join(cwd, "node_modules", "graphql-codegen-typescript-client", "dist", "index.js");
+        if (await fs.pathExists(codegenTemplate)) {
+            let contents = (await fs.readFile(codegenTemplate)).toString();
+            contents = contents.replace(
+                /{{ name }}: {{ convertedFieldType/i,
+                "{{ name }}?: {{ convertedFieldType");
```
Comment (ddgenome) [inline bin/gql-gen.ts]: If this regular expression is case insensitive, shouldn't the replacement maintain the same case?

```typescript
                /{{ (name) }}: {{ (convertedfieldtype)/i,
                "{{ $1 }}?: {{ $2");
```

File: bin/gql-gen.ts [hunk 2]
```diff
@@ -40,53 +40,54 @@ async function main(): Promise<void> {
         const cwd = process.cwd();
         const lib = libDir(cwd);
 
+        // patch up codegen template to create optional properties
+        const codegenTemplate = path.join(cwd, "node_modules", "graphql-codegen-typescript-client", "dist", "index.js");
+        if (await fs.pathExists(codegenTemplate)) {
+            let contents = (await fs.readFile(codegenTemplate)).toString();
```
Comment (ddgenome) [inline bin/gql-gen.ts]: Should all this be in a try/catch block?

File: bin/gql-gen.ts [hunk 3]
```diff
@@ -40,53 +40,54 @@ async function main(): Promise<void> {
...
+
         const graphqlFiles = await util.promisify(glob)(graphQlGlob);
+
         if (graphqlFiles && graphqlFiles.length > 0) {
-            gqlGenArgs.push(graphQlGlob);
+            config.documents = [graphQlGlob];
+            await generate(config);
         } else {
-            console.info("No GraphQL files found in project, generating default types");
+            console.info("No GraphQL files found in project. Skipping type generation...");
```
Comment (ddgenome) [inline bin/gql-gen.ts]: Are the default types brought in via a dependency?

File: bin/gql-gen.ts (hunk 2 above)
Comment (cdupuis) [reply bin/gql-gen.ts]: Not sure. I’d like it to fail if this fails.

File: bin/gql-gen.ts (hunk 1 above)
Comment (cdupuis) [reply bin/gql-gen.ts]: I’ll remove the i in the regex.

File: bin/gql-gen.ts (hunk 3 above)
Comment (cdupuis) [reply bin/gql-gen.ts]: What default types? Without any graphql files, what types would you expect?

**Comment by cdupuis** [comment]
@ddgenome any further thoughts on this?

**Comment by cdupuis** [comment]
@ddgenome any more on this?

File: bin/gql-gen.ts (hunk 3 above)
Comment (ddgenome) [reply bin/gql-gen.ts]: All the core types in the graph. Typically when there are no custom types in the project, you still get a types.ts with all core types Atomist defines. Will those be available some other way.

**Comment by ddgenome** [comment]
Sorry, this slipped off my radar. I clarified my question.

File: bin/gql-gen.ts (hunk 3 above)
Comment (cdupuis) [reply bin/gql-gen.ts]: No. They won’t.

Comment (cdupuis) [reply bin/gql-gen.ts]: You only get a generated types.ts when you actually define graphql queries or subscriptions. At that point you get a types.ts with the core types you used in your queries/subscriptions.

Comment (ddgenome) [reply bin/gql-gen.ts]: I think I am confusing custom schema and GraphQL files. You still create a types.ts if there is no custom schema, but you do not if there are no GraphQL queries/subscriptions/etc. Sorry for the confusion.

**Pull request merged**